### ISSUE TITLE
Block names cannot be integers

### ISSUE THREAD

**Issue by stinkysGTI** [trendwerk/multiple-content-blocks#62]
I was trying to figure out where to put the code to only have it apply to posts and in doing so, I created multiple fields that I didn't need, and with no obvious way to delete them, I brought up the database in phpmyadmin and did a search for "%_mcb%". I then deleted all of the entries since it just seemed as though they were the fields created and the content in said fields.

I added just the one field I needed and everything was fine until I tried to update the page. It seems to still write to the database, but I get this error:

```
Warning: array_map() [function.array-map]: Argument #2 should be an array in /home/adoptall/public_html/wp-includes/post.php on line 2127
```

I then tried a clean start, so I deleted the newly created entries and uninstalled the plugin. Then I installed it again, but I'm getting the same result.

Any ideas why this is happening?

**Comment by haroldangenent** [comment]
Blocks that are unused in a template will become inactive (automatically). You can turn on "Show inactive blocks" if you want to remove them completely from the database, but this is not necessary.

As far as I understand, your blocks didn't become inactive. Is that right?

When you turn off the plugin, do you still get the PHP warning?

**Comment by stinkysGTI** [comment]
You're correct, they were not inactive.

As of now, they appear as hoped on the back end; just one block per post. However, I get that error only when the plugin is activated. There's also a "headers were already sent" error that comes after the previously stated error. It started after i deleted the MCB database entries manually, but I'm not sure why that would have effected anything though.

Any thoughts on how I could get it back to a clean start? I guess I should've backed up the database before making edits.

**Comment by haroldangenent** [comment]
I'm not sure what's happening on your install. Did you manage to fix this?

Can you reproduce this in a clean install? Because there might be something else going on. I'd love to hear steps to reproduce this.

**Comment by stinkysGTI** [comment]
I was able to reproduce it without messing with the database at all this time.

I created a new subdomain and database so everything was fresh. I installed wordpress and the multiple content block plugin. Then I added the code below to a template file I created.

```
		$name = get_the_ID();
		echo '<!-- spanish -->';
		the_block( $name, array(
			'label'         => __('Spanish Version', 'text-domain'),
			'type'          => 'editor'
		));
```

When I load the "Sample Page" page and look at the back end, the block has been added, but i get the following error when trying to update the content:
```
Warning: array_map() [function.array-map]: Argument #2 should be an array in /wp-includes/post.php on line 2127

Warning: Cannot modify header information - headers already sent by (output started at /wp-includes/post.php:2127) in /wp-admin/post.php on line 233

Warning: Cannot modify header information - headers already sent by (output started at /wp-includes/post.php:2127) in /wp-includes/pluggable.php on line 1178
```

It still saves it though, it just breaks right after doing so.

Perhaps I inserted the code in the wrong place? I installed WP 4.1 (currently the newest version)

If I remove the code but keep the plugin active, the block is still there, but I also still get the error. It only goes away if I deactivate it. I also tried adding it to a default template file in case I had something custom that was causing conflicts.

Weird right?

**Comment by haroldangenent** [comment]
Thanks for investigating.

Are you able to reproduce this when you use a string as the name for the block? (e.g. `the_block( 'spanish' )` instead of `the_block( get_the_ID() )`)

**Comment by stinkysGTI** [comment]
Yes unfortunately. I'm not sure what else to try

**Comment by haroldangenent** [comment]
Are you sure? I could definitely reproduce this error when using integers as the block name, but when using normal strings I didn't get the `array_map()` error.

If you're in trouble on a production server, I suggest suppressing errors (wouldn't want to show them on production). For example (in your config file):

```
define( 'WP_DEBUG', false );
error_reporting( 0 );
ini_set( 'display_errors', 0 );
```

But we should still investigate this further.

**Comment by stinkysGTI** [comment]
Maybe because it was already corrupted by the integer being passed? I'll mess with it some more today. I'm going to start fresh again and see what happens.

**Comment by stinkysGTI** [comment]
It seems as though an integer breaks and corrupts it somehow. I used a string first and there were no errors. Then I tried using strval(get_the_ID()) hoping to set it as a string and I got the errors. Then when I reverted back to a static string, it still threw the errors

**Comment by stinkysGTI** [comment]
My only problem now is on the original site, it breaks when I try to revert back to a string even when I delete all entries in the database containing "_mcb". Any idea how to clear it so I can reinstall the plugin fresh?

**Comment by haroldangenent** [comment]
Thanks for your work. It's so strange that it keeps breaking whenever you revert back to a string. Maybe try suppressing errors for your specific case? Did you try to remove `_mcb-blocks` as well?

**Comment by stinkysGTI** [comment]
I didn't try suppressing errors, but what I did do was open up the database in phpmyadmin and searched for `_mcb`. Then edited the `_mcb_blocks` meta_value entry and removed just that block entry (I had 2 on the same post). Then as I was going through the rest of the search results I also deleted the extra content row as well. There were 2 post_id's that were the same with different content.

Before:
a:2:{i:6;a:2:{s:5:"label";s:15:"Spanish Version";s:4:"type";s:6:"editor";}s:7:"spanish";a:2:{s:5:"label";s:15:"Spanish Version";s:4:"type";s:6:"editor";}}

After:
a:2:{i:6;a:2:{s:5:"label";s:15:"Spanish Version";s:4:"type";s:6:"editor";}}

It's like it was trying to save to a post_id of "spanish" or something instead of an integer.

That actually seemed to fix it. So now I'm going through the original, live site where I initially broke it and deleting all the blocks. Everything seems fine now.
